content/spin/kv-store.md

```diff
@@ -0,0 +1,355 @@
+title = "Using Key Value Storage with Spin Applications"
+template = "spin_main"
+date = "2023-02-21T00:00:00Z"
+enable_shortcodes = true
+[extra]
+
+---
+- [Using Key Value Storage With Spin Applications](#using-key-value-storage-with-spin-applications)
+    - [Prerequisites](#prerequisites)
+    - [Use Spin SDK To Open And Persist Data In A Default Key/Value Store](#use-spin-sdk-to-open-and-persist-data-in-a-default-keyvalue-store)
+    - [Configure `spin.toml` To Use A Default Key/Value Store](#configure-spintoml-to-use-a-default-keyvalue-store)
+    - [Build And Deploy Your Spin App](#build-and-deploy-your-spin-app)
+    - [Store and Retrieve Data From Your Default Key/Value Store](#store-and-retrieve-data-from-your-default-keyvalue-store)
+    - [Conclusion](#conclusion)
+    - [Next Steps](#next-steps)
+
+## Using Key Value Storage with Spin Applications
+
+ Spin application are best suited for event-driven, stateless workloads that have low-latency requirements. If your Spin applications needs to persist state, the data will need to be stored in an external tool or service. In [Spin v0.9.0](https://www.fermyon.com/blog/spin-v09), we've added support for key/value storage. This is a storage option for applications that need to data in the form of key/value pairs and are satistifed by a BASE consistency model. Workload examples include general value caching, session caching, counters, serialized application state.
+
+ Spin users can use new a API in the Spin SDKs for persisting and retrieving non-relational data from a key/value store across multiple requests to and from the same application. Together with the updated SDKs we are introducing a built-in, _local_ key/value store available with minimal configuration for every Spin application. The default key/value store is available for the following SKDs: Go, JavaScript/TypeScript, and Rust. 
+
+ Let's get started with creating and deploying your first Spin application that uses key/value storage.
+
+## Prerequisites
+
+First, follow [this guide](./install.md) to install Spin v0.9.0 or later. To ensure you have the correct version, you can check with this command:
+
+<!-- @selectiveCpy -->
+
+```bash
+spin --version
+```
+
+## Use Spin SDK To Open And Persist Data In A Default Key/Value Store 
+
+First, we must open a default key/value store for our Spin application. This is a special store that every environment running Spin applications will make available for their application (which when running Spin applications locally, is built using [SQLite](https://www.sqlite.org/index.html)). Then we will complete the following actions: 
```

Comment:
I think this is a good opportunity to state that this store is available only to this application's components. For this reason, it may be best to start with the `Spin.toml` rather than the guest code so we can explain that it is one store per app shared by all components that are granted access

Reply:
That's a great point. We now have a newly formatted section Re: "multiple requests to and from the same application".
Also added the `spin.toml` configuration file before the other application source code.